hedy.py

```diff
@@ -1932,31 +1939,56 @@ def play(self, meta, args):
     def assign(self, meta, args):
         variable_name = args[0]
         value = args[1]
-
-        if self.is_random(value) or self.is_list(value):
-            exception = self.make_index_error_check_if_list([value])
-            return exception + variable_name + " = " + value + self.add_debug_breakpoint()
+        if not self.microbit:
+            if self.is_random(value) or self.is_list(value):
+                exception = self.make_index_error_check_if_list([value])
+                return exception + variable_name + " = " + value + self.add_debug_breakpoint()
+            else:
+                if self.is_variable(value, meta.line):  # if the value is a variable, this is a reassign
+                    value = self.process_variable(value, meta.line)
+                    return variable_name + " = " + value + self.add_debug_breakpoint()
+                else:
+                    # if the assigned value is not a variable and contains single quotes, escape them
+                    value = process_characters_needing_escape(value)
+                    return variable_name + " = '" + value + "'" + self.add_debug_breakpoint()
         else:
-            if self.is_variable(value, meta.line):  # if the value is a variable, this is a reassign
-                value = self.process_variable(value, meta.line)
-                return variable_name + " = " + value + self.add_debug_breakpoint()
+            if self.is_random(value) or self.is_list(value):
+                exception = self.make_index_error_check_if_list([value])
+                return "    " + exception + str(variable_name) + " = " + value + self.add_debug_breakpoint()
```

Comment:
This line is problematic because the transpiled code of the exception is multiple lines (at least 4). The make_index_error_check_if_list method does not have any microbit logic, so this will just offset the first line of the snippet.

Perhaps I am missing context here but I wonder why with microbit the transpiled commands start with spacing? I suspect this may cause big issues when we transpile functions and definitions at a later stage (if this is in scope). Would it be an option to transpile and then indent the whole program?

Reply:
As we discussed in our meeting i was adding white spaces in the transpiler so that the output can be correctly formatted for micro-bit. However, we chose a different approach. The white spaces if needed will be handled in app.py in the save_transpiled_code_for_microbit function.